core/src/main/java/com/microsoft/applicationinsights/internal/heartbeat/HeartBeatProvider.java

```diff
@@ -0,0 +1,304 @@
+        payload.setPayloadValue(propertyValue);
+        heartbeatProperties.put(propertyName, payload);
+        setResult = true;
+
+      }
+      catch (Exception e) {
+        InternalLogger.INSTANCE.warn("failed to set heartbeat property name %s, value %s, "
+            + "stack trace is: %s", propertyName, propertyValue, ExceptionUtils.getStackTrace(e));
+      }
+    }
+    else {
+      InternalLogger.INSTANCE.warn("cannot set property without property name");
+    }
+    return setResult;
+  }
+
+  @Override
+  public boolean isHeartBeatEnabled() {
+    return isEnabled;
+  }
+
+  @Override
+  public void setHeartBeatEnabled(boolean isEnabled) {
+    this.isEnabled = isEnabled;
+  }
+
+  @Override
+  public List<String> getExcludedHeartBeatPropertyProviders() {
+    return this.disabledHeartBeatPropertiesProviders;
+  }
+
+  @Override
+  public void setExcludedHeartBeatPropertyProviders(
+      List<String> excludedHeartBeatPropertyProviders) {
+    this.disabledHeartBeatPropertiesProviders = excludedHeartBeatPropertyProviders;
+  }
+
+  @Override
+  public long getHeartBeatInterval() {
+    return this.interval;
+  }
+
+  @Override
+  public void setHeartBeatInterval(long timeUnit) {
+    // user set time unit in seconds
+    if (timeUnit <= HeartBeatProviderInterface.MINIMUM_HEARTBEAT_INTERVAL) {
+      this.interval = HeartBeatProviderInterface.MINIMUM_HEARTBEAT_INTERVAL;
+    }
+    else {
+      this.interval = timeUnit;
+    }
+  }
+
+  @Override
+  public List<String> getExcludedHeartBeatProperties() {
+    return this.disableDefaultProperties;
+  }
+
+  @Override
+  public void setExcludedHeartBeatProperties(List<String> excludedHeartBeatProperties) {
+    this.disableDefaultProperties = excludedHeartBeatProperties;
+  }
+
+  @Override
+  public void stop(long timeout, TimeUnit timeUnit) {
+    propertyUpdateService.shutdown();
+    heartBeatSenderService.shutdown();
+    try {
+      propertyUpdateService.awaitTermination(timeout, timeUnit);
+      heartBeatSenderService.awaitTermination(timeout, timeUnit);
+    }
+    catch (InterruptedException e) {
+      InternalLogger.INSTANCE.warn("unable to successfully terminate heartbeat module, "
+          + "encountered and exception with stacktrace, %s", ExceptionUtils.getStackTrace(e));
+    }
+  }
+
+  /**
+   * Send the heartbeat item synchronously to application insights backend.
+   */
+  private void send() {
+
+    synchronized (lock) {
+      MetricTelemetry telemetry = (MetricTelemetry)gatherData();
+      telemetry.getContext().getOperation().setSyntheticSource(HEARTBEAT_SYNTHETIC_METRIC_NAME);
+      telemetryClient.trackMetric(telemetry);
+      heartbeatsSent.incrementAndGet();
+      InternalLogger.INSTANCE.trace("No of heartbeats sent, %s", heartbeatsSent.get());
+    }
+
+  }
+
+  /**
+   * Creates and returns the heartbeat telemetry.
+   * @return Metric Telemetry which represent heartbeat.
+   */
+  private Telemetry gatherData() {
```

Comment:
This should just return `MetricTelemetry` so you don't have to cast it.

Reply:
Fair, Okay this should be addressed.